Source/Core/DolphinQt2/Debugger/MemoryWidget.cpp

```diff
@@ -0,0 +1,583 @@
+    return;
+  }
+
+  if (m_find_ascii->isChecked())
+  {
+    std::string ascii = m_data_edit->text().toStdString();
+
+    for (char c : ascii)
+      PowerPC::HostWrite_U8(static_cast<u8>(c), addr++);
+  }
+  else
+  {
+    bool good_value;
+    u64 value = m_data_edit->text().toULongLong(&good_value, 16);
+
+    if (!good_value)
+    {
+      QMessageBox::critical(this, tr("Error"), tr("Bad value provided."));
+      return;
+    }
+
+    if (value == static_cast<u8>(value))
+    {
+      PowerPC::HostWrite_U8(static_cast<u8>(value), addr);
+    }
+    else if (value == static_cast<u16>(value))
+    {
+      PowerPC::HostWrite_U16(static_cast<u16>(value), addr);
+    }
+    else if (value == static_cast<u32>(value))
+    {
+      PowerPC::HostWrite_U32(static_cast<u32>(value), addr);
+    }
+    else
+      PowerPC::HostWrite_U64(value, addr);
+  }
+
+  Update();
+}
+
+static void DumpArray(const std::string& filename, const u8* data, size_t length)
+{
+  if (!data)
+    return;
+
+  File::IOFile f(filename, "wb");
+  f.WriteBytes(data, length);
+}
+
+void MemoryWidget::OnDumpMRAM()
+{
+  DumpArray(File::GetUserPath(F_RAMDUMP_IDX), Memory::m_pRAM, Memory::REALRAM_SIZE);
+}
+
+void MemoryWidget::OnDumpExRAM()
+{
+  if (SConfig::GetInstance().bWii)
+  {
+    DumpArray(File::GetUserPath(F_ARAMDUMP_IDX), Memory::m_pEXRAM, Memory::EXRAM_SIZE);
+  }
+  else
+  {
+    DumpArray(File::GetUserPath(F_ARAMDUMP_IDX), DSP::GetARAMPtr(), DSP::ARAM_SIZE);
+  }
+}
+
+void MemoryWidget::OnDumpFakeVMEM()
+{
+  DumpArray(File::GetUserPath(F_FAKEVMEMDUMP_IDX), Memory::m_pFakeVMEM, Memory::FAKEVMEM_SIZE);
+}
+
+std::vector<u8> MemoryWidget::GetValueData()
+{
+  std::vector<u8> search_for;  // Series of bytes we want to look for
+
+  if (m_find_ascii->isChecked())
+  {
+    std::string s = m_data_edit->text().toStdString();
+
+    for (char c : s)
+      search_for.push_back(c);
+  }
+  else
+  {
+    bool good;
+    u64 value = m_data_edit->text().toULongLong(&good, 16);
+
+    if (!good)
+    {
+      return {};
+    }
+
+    if (value == static_cast<u8>(value))
+    {
+      search_for.push_back(static_cast<u8>(value));
+    }
+    else if (value == static_cast<u16>(value))
+    {
+      search_for.push_back(value & 0xFF);
+      search_for.push_back((value >> 8) & 0xFF);
+    }
+    else if (value == static_cast<u32>(value))
+    {
+      search_for.push_back(value & 0xFF);
+      search_for.push_back((value >> 8) & 0xFF);
+      search_for.push_back((value >> 16) & 0xFF);
+      search_for.push_back((value >> 24) & 0xFF);
+    }
+    else
+    {
+      search_for.push_back(value & 0xFF);
+      search_for.push_back((value >> 8) & 0xFF);
+      search_for.push_back((value >> 16) & 0xFF);
+      search_for.push_back((value >> 24) & 0xFF);
+
+      search_for.push_back((value >> 32) & 0xFF);
+      search_for.push_back((value >> 40) & 0xFF);
+      search_for.push_back((value >> 48) & 0xFF);
+      search_for.push_back((value >> 52) & 0xFF);
```

Comment:
`search_for.push_back((value >> 56) & 0xFF);`

![snapcrab_memory_2018-3-24_16-55-21_no-00](https://user-images.githubusercontent.com/21278142/37861788-2cc5d086-2f84-11e8-995c-8ff8f1af9e93.png)
The value is byte-swapped when searching. Set Value works correctly.

Reply:
Should work properly now.